verifiers/internal/gha/slsaprovenance/v1.0/provenance.go

```diff
@@ -112,29 +113,41 @@ func (prov *ProvenanceV1) Subjects() ([]intoto.Subject, error) {
 }
 
 func (prov *ProvenanceV1) GetBranch() (string, error) {
-	// TODO(https://github.com/slsa-framework/slsa-verifier/issues/472): Add GetBranch() support.
```

Comment:
so this will remove support for all cases, not just the cases where checkout sha != github event sha. Correct? Would this not be a breaking change? (same question for getTag()).

Reply:
I think in all cases the "most correct" source branch and source tag will come from the materials/resolvedDependencies. My understanding is that up until we added the `sha1` input all of our builders have provided the same value in the parameters as in the dependencies. So the value returned here shouldn't change in that case.

So I believe that it shouldn't break older provenance, but I suppose that technically we didn't actually check that the branch and sha were the same between the source and trigger.

I suppose the alternative is to get from `InternalParameters` if the shas are the same and the source uri if they are different but I'm not sure if we gain anything by doing that. I'll look a bit into what the older versions of the builders did.

Reply:
branch and tag info is not present in materials / resolvedDeps. I think it's only present via the GH context?

Reply:
There is a branch/tag ref on the URI in the materials/resolvedDeps. We explicitly remove it when sha1 is specified.